keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -104,7 +108,54 @@ class ApplicationService(
     }
   }
 
-  private fun getFirstDeliveryConfigFor(application: String): DeliveryConfig? =
+  /**
+   * Takes an artifact version, plus information about the type of artifact, and constructs a summary view.
+   * This should be supplemented/re-written to use actual data from stash/git/etc instead of parsing everything
+   * from the version string.
+   */
+  private fun versionToSummary(
+    artifact: DeliveryArtifact,
+    version: String,
+    environments: Set<ArtifactSummaryInEnvironment>
+  ): ArtifactVersionSummary =
+    when (artifact.type) {
+      deb -> {
+        val appversion = AppVersion.parseName(version)
+        ArtifactVersionSummary(
+          version = version,
+          environments = environments,
+          displayName = appversion?.version ?: version.removePrefix("${artifact.name}-"),
+          build = if (appversion != null) BuildMetadata(id = appversion.buildNumber.toInt()) else null,
+          git = if (appversion != null) GitMetadata(commit = appversion.commit) else null
+        )
+      }
+      docker -> {
+        var build: BuildMetadata? = null
+        var git: GitMetadata? = null
+        val dockerArtifact = artifact as DockerArtifact
+        if (dockerArtifact.hasBuild()) {
+          // todo eb: this could be less brittle
+          val regex = Regex("""^.*-h(\d+).*$""")
+          val result = regex.find(version)
+          if (result != null && result.groupValues.size == 2) {
+            build = BuildMetadata(id = result.groupValues[1].toInt())
+          }
+        }
+        if (dockerArtifact.hasCommit()) {
+          // todo eb: this could be less brittle
+          git = GitMetadata(commit = version.substringAfterLast("."))
+        }
+        ArtifactVersionSummary(
+          version = version,
+          environments = environments,
+          displayName = version,
+          build = build,
```

Comment:
If we have the build id here, we can link it to the logs within the ci tab (I guess we will need to do it in -nflx only?)

Reply:
once we have enough details to put together a full build link (so controller + build name + number) I intend to do a -nflx only link to our CI view as you suggest. Currently all we have is the number that's already in the version string, so unfortunately I can't choose the right build name to look for yet.